### PR TITLE
[ML] Use results retention time for deleting system annotations

### PR DESCRIPTION
**Pull request by droberts195**
In #75617 a new setting, system_annotations_retention_days, was
added to control how long system annotations are retained for.
We now feel that this setting is redundant and that system
annotations should be retained for the same period as results.
This is intuitive and defensible, as system annotations can be
considered a type of result.

Followup to #75617

### CODE REVIEW

**Comment by droberts195** [comment]
`>non-issue` because the original change was never released.

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -1433,16 +1433,6 @@ bucket result are deleted from {es}. The default value is null, which means all
 results are retained.
```
Comment (benwtrent) [inline docs/reference/ml/ml-shared.asciidoc:1433]: We should comment here that system created annotations are also deleted.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdate.java [hunk 2]
```diff
@@ -193,7 +192,9 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalLong(modelSnapshotRetentionDays);
         out.writeOptionalLong(dailyModelSnapshotRetentionAfterDays);
         out.writeOptionalLong(resultsRetentionDays);
-        out.writeOptionalLong(systemAnnotationsRetentionDays);
+        if (out.getVersion().onOrAfter(Version.V_7_15_0) && out.getVersion().before(Version.V_8_0_0)) {
+            out.writeOptionalLong(null);
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdate.java:196]: I think *technically* this should be `out.writeOptionalLong(resultsRetentionDays);`

But, that doesn't matter really as when this change is backported this branch is removed.

**Review by benwtrent — APPROVED** [review]
very minor things, but I like the change. Keep it simple!

**Pull request merged**